### ISSUE TITLE
[<Ray component: Tune>] TensorflowCheckpoint Temp Folder is hard coded to system variable

### ISSUE THREAD

**Issue by drukoz** [ray-project/ray#41205]
### Description

We need to change this as this way we can set where we ant to save the tmp files that are used while trials are paused .
we have a HPC that has a small drive and a huge network storage we had to change the tmp folder for the windows env and i am guessing that linux has the same issue as well when this can be configured by python env variable or passing it to the class.

this is a copy of the ticket issue : 
https://github.com/ray-project/ray/issues/40956

the reason why i am reopening this is because the given solution does not work and the tmp folder is still being filled up and left with models that are not deleted. i think the suggestion i had to set the tmp folder as a variable was a better idea over all. either that or fix the issue with the code changes that were made.

### Use case

Configure the directory the temporary "model.keras" which are of a decent size is saved when using raytune tensorflow checkpoint or any derivative like ReportCheckpointCallback for tensorflow

**Comment by matthewdeng** [comment]
Hey what version of Ray are you using. https://github.com/ray-project/ray/pull/41033 is included in the nightly wheel and should delete the temp folder

**Comment by drukoz** [comment]
i am using the version 3.0.0 using specific commit 036439609cda989407bb8a35902418876904d8c4 which is from last night it is still not deleting the tmp folders i double checked the local code file for keras and the changes are there! i am running an model with 1000 samples and the tmp files from the beginning of the run are still in the folder it filled up the whole space even though there are checkpoints being made




`    model.fit(train_data, batch_size=config["batch_size"] ,epochs=120,validation_data=val_data,verbose=False,  
              callbacks=[ReportCheckpointCallback(metrics={"mean_accuracy": "accuracy","mean_loss":"loss", "val_loss":"val_loss","val_acc":"val_accuracy"})])
`

this is the code i have for the report checkpoint callback

**Comment by drukoz** [comment]
i think the issue is for when the training is happening and the models are filling up the tmp folder

**Comment by matthewdeng** [comment]
@drukoz can you share a quick repro of what you're seeing?

**Comment by drukoz** [comment]
Hello,
Sorry i have not been working on this project for a couple weeks
here are the commands i am using so that you guys can repo this

on headnode: 
ray start --head --port=6379 --node-ip-address=192.168.1.210 --temp-dir="C:\Users\Administrator\AppData\Local\Temp\2"

on worker node:
ray start --address=192.168.1.210:6379

then on the server node i run our ray tune script with this configuration for storage

`run_config=train.RunConfig(
            name="REMOTETRY",
            storage_path="F:\\NFS\\",
        ),`

that storage path is the NFS shared path from the head node where we have 58 TB of storage 
![image](https://github.com/ray-project/ray/assets/17682552/7fbd218f-5c97-47d5-a208-18403c04f3e4)

currently there are 65 terminated trials and the worker node as well as the headnode has over 160 tmp folders still with models filling up the space as the worker node only has 256GB of storage

**Comment by MauriceBrg** [comment]
We're observing the same problem with Ray 2.9.0, we're running a single trial and the checkpoints end up in /tmp where they're not being deleted despite setting `num_to_keep` to a low number.
